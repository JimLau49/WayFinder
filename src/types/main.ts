export interface Location {
  latitude: number;
  longitude: number;
}

export interface Region extends Location {
  latitudeDelta: number;
  longitudeDelta: number;
}

export interface Building {
  id: BuildingId;
  campusId: CampusId;
  displayName: string;
  address: string;
  location: Location;
  departments: LinkItem[];
  services: LinkItem[];
  boundingBox: Location[];
}

export interface LinkItem {
  id: number;
  title: string;
  link: string;
}

export interface Campus {
  id: CampusId;
  displayName: string;
  description: string;
  region: Region;
  buildings: BuildingId[];
}

export interface MarkerLocation {
  id: string;
  displayName: string;
  location: Location;
}

export interface IndoorLocation extends MarkerLocation {
  buildingId: BuildingId;
  level: number;
}

export interface POI extends IndoorLocation {
  description: string;
  category: POICategory;
}

export interface Connector extends POI {
  type: ConnectorType;
  connections: Connection[];
}

export interface Connection {
  level: number;
  location: Location;
}

export interface BuildingFloor {
  id: number;
  buildingId: BuildingId;
  level: number;
  bounds: [Coordinate, Coordinate];
  image: any;
  travelNodes: TravelNode[];
}

export interface TravelNode {
  id: number;
  location: Location;
  children: number[];
}

export interface IndoorInformation {
  currentFloor: IndoorFloor;
  floors: IndoorFloor[];
}

export interface IndoorFloor {
  level: number;
  index: number;
}

export interface Range {
  min: number;
  max: number;
}

export type PQItem = { id: number; parent: number; g: number };

export type TravelEdge = [TravelNode, TravelNode];

export type Line = [Location, Location];

export type Coordinate = [number, number];

export enum CampusId {
  SGW = "SGW",
  Loyola = "Loyola"
}

export enum BuildingId {
  // SGW Campus
  B,
  CB,
  CI,
  CL,
  D,
  EN,
  ER,
  EV,
  FA,
  FB,
  FG,
  GA,
  GM,
  GN,
  GS,
  H,
  K,
  LB,
  LD,
  LS,
  M,
  MB,
  MI,
  MU,
  P,
  PR,
  Q,
  R,
  RR,
  S,
  SB,
  T,
  TD,
  V,
  VA,
  X,
  Z,
  // Loyola Campus
  AD,
  BB,
  BH,
  CC,
  CJ,
  DO,
  FC,
  GE,
  HA,
  HB,
  HC,
  HU,
  JR,
  PC,
  PS,
  PT,
  PY,
  QA,
  RA,
  RF,
  SC,
  SH,
  SI,
  SP,
  TA,
  VE,
  VL
}

export enum ZoomLevel {
  // Too far
  NONE,
  // Really zoomed out
  CAMPUS,
  // Medium
  OUTDOOR,
  // Close
  INDOOR
}

export enum POICategory {
  Classroom,
  Washroom,
  Restaurant,
  Association,
  Elevator,
  Escalator,
  Stairs,
  Exit,
  Service
}

export enum ConnectorType {
  Elevator,
  Stairs,
  Ramp,
  Escalator
}

export enum TravelState {
  NONE,
  PLANNING,
  TRAVELLING
}
