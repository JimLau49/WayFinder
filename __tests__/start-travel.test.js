import React from "react";
import StartTravel from "../src/components/search/start-travel.component";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { TravelState } from "../src/types/main";

describe("StartTravel component", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<StartTravel />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call provided function on press", () => {
    const mockSetTravelState = jest.fn();
    const mockOnStartTravelPlan = jest.fn();
    const wrapper = shallow(
      <StartTravel
        setTravelState={mockSetTravelState}
        onStartTravelPlan={mockOnStartTravelPlan}
      />
    );
    const touchableOpacity = wrapper.find("TouchableOpacity");
    touchableOpacity.simulate("press");
    expect(mockSetTravelState).toHaveBeenCalledTimes(1);
    expect(mockOnStartTravelPlan).toHaveBeenCalledTimes(1);
    expect(mockSetTravelState.mock.calls[0][0]).toBe(TravelState.TRAVELLING);
  });
});
