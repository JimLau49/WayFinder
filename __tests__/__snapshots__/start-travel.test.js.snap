// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartTravel component should match snapshot 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "flex": 1,
      "height": 60,
      "position": "absolute",
      "right": 20,
      "top": 1234,
      "width": 60,
      "zIndex": 1,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "height": 80,
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#AA2B45",
            "borderColor": "rgba(0,0,0,0.2)",
            "borderRadius": 50,
            "borderWidth": 1,
            "flex": 1,
            "height": 60,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 60,
          }
        }
      >
        <Text />
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;
